src/transformers/utils/model_parallel_utils.py

```diff
@@ -12,10 +12,186 @@
+            Col("q_proj", "k_proj", "v_proj", "fc1"),
+            Row("out_proj", "fc2"),
+            Update("embed_dim", "num_heads"),
+        ],
+        Bert=[
+            Col("query", "key", "value", "intermediate.dense"),
+            Row("output.dense"),
+            Update("num_attention_heads", "all_head_size"),
+        ],
+        T5=[
+            Col("q", "k", "v", "DenseReluDense.wi"),
+            Row("o", "DenseReluDense.wo", "relative_attention_bias"),
+            Update("d_model", "n_heads", "inner_dim"),
+        ],
+        GPT2=[
+            Col("c_attn", reverse=True, combined_qkv=True),
+            Col("c_fc", "q_attn", reverse=True),
+            Row("c_proj", reverse=True),
+            Update("embed_dim", "split_size", "num_heads"),
+        ],
+        GPTNeo=[
+            Col("q_proj", "k_proj", "v_proj", "c_fc"),
+            Row("out_proj", "c_proj"),
+            Update("embed_dim", "num_heads"),
+        ],
+        GPTJ=[
+            Col("q_proj", "k_proj", "v_proj", "fc_in"),
+            Row("out_proj", "fc_out"),
+            Update("embed_dim", "num_attention_heads"),
+        ],
+        Electra=[
+            Col("query", "key", "value", "intermediate.dense"),
+            Row("output.dense"),
+            Update("num_attention_heads", "all_head_size"),
+        ],
+        Roberta=[
+            Col("query", "key", "value", "intermediate.dense"),
+            Row("output.dense"),
+            Update("num_attention_heads", "all_head_size"),
+        ],
+    )
+
+    def __init__(self):
+        cache_tp_mapping = {}
+
+        for cls_name, mapping in self.__MAPPING__.items():
+            cls = self._load_class_by_model_name(cls_name)
+            cache_tp_mapping[cls] = []
+
+            for elem in mapping:
+                for name in elem.name:
+                    copy_elem = copy.deepcopy(elem)
+                    copy_elem.name = name
+                    cache_tp_mapping[cls].append(copy_elem)
+
+        self.__MAPPING__ = {cls: {} for cls in cache_tp_mapping}
+        # clear exist mapping rather than making new mapping dict
+
+        for cls, mapping in cache_tp_mapping.items():
+            for elem in mapping:
+                if elem.code in self.__MAPPING__[cls]:
+                    self.__MAPPING__[cls][elem.code].append(elem)
+                else:
+                    self.__MAPPING__[cls][elem.code] = [elem]
+
+    @staticmethod
+    def _load_class_by_model_name(model_name):
+        transformers = importlib.import_module("transformers")
+        cls = getattr(transformers, f"{model_name}PreTrainedModel", None)
+        if cls is None:
+            cls = getattr(transformers, f"{model_name}PretrainedModel", None)
+        assert cls is not None, f"Can not import the model named {cls}."
+        return cls
+
+    def get_mapping(self, model):
+        for cls, mapping in self.__MAPPING__.items():
+            if isinstance(model, cls):
+                return mapping
```

Comment:
If we use `defaultdict`, we might want to re-wrap `mapping` around `dict` if we want to prevent `mapping["Row"]` from returning an empty list, for instance.

```suggestion
                return mapping
```